src/baldr/graphtile.cc

```diff
@@ -449,33 +427,35 @@ GraphTile::FileSuffix(const GraphId& graphid, const std::string& fname_suffix, b
                           ? TileHierarchy::GetTransitLevel()
                           : TileHierarchy::levels()[graphid.level()];
 
-  // figure out how many digits
-  auto max_id = level.tiles.ncolumns() * level.tiles.nrows() - 1;
+  // figure out how many digits in tile-id
+  const auto max_id = level.tiles.ncolumns() * level.tiles.nrows() - 1;
   size_t max_length = static_cast<size_t>(std::log10(std::max(1, max_id))) + 1;
   const size_t remainder = max_length % 3;
   if (remainder) {
     max_length += 3 - remainder;
   }
+  assert(max_length % 3 == 0);
 
-  // make a locale to use as a formatter for numbers
-  std::ostringstream stream;
-  if (is_file_path) {
-    stream.imbue(dir_locale);
-  } else {
-    stream.imbue(url_locale);
-  }
+  // Calculate tile-id string length with separators
+  const size_t tile_id_strlen = max_length + max_length / 3;
+  assert(tile_id_strlen % 4 == 0);
+
+  const char separator = is_file_path ? filesystem::path::preferred_separator : '/';
 
-  // if it starts with a zero the pow trick doesn't work
-  if (graphid.level() == 0) {
-    stream << static_cast<uint32_t>(std::pow(10, max_length)) + graphid.tileid() << fname_suffix;
-    std::string suffix = stream.str();
-    suffix[0] = '0';
-    return suffix;
+  std::string tile_id_str(tile_id_strlen, '0');
+  size_t ind = tile_id_strlen - 1;
+  for (uint32_t tile_id = graphid.tileid(); tile_id != 0; tile_id /= 10) {
+    tile_id_str[ind--] = '0' + static_cast<char>(tile_id % 10);
+    if ((tile_id_strlen - ind) % 4 == 0) {
+      tile_id_str[ind--] = separator;
+    }
+  }
+  // add missing separators
+  for (size_t sep_ind = 0; sep_ind < ind; sep_ind += 4) {
+    tile_id_str[sep_ind] = separator;
   }
```

Comment:
im not sure you actually need two loops here. i think if you just change to while part of the loop to be `ind >= 0` it should still work. feel free to ignore this suggestion. otherwise the code looks good. thanks for simplifying this

Reply:
I tried to do it with one loop, but then rewrite it with two loops. And imho it looks clearer this way, putting zeros to the places when they are already seemed not good to me.

Reply:
It's possible to simplify this code with initializing `tile_id_str` with `"/000/000/.../000"` with maximum possible size, but I'm not sure what is maximum size here.